agent/recorder.go

```diff
@@ -75,9 +79,14 @@ func NewSpanRecorder(agent *Agent) *SpanRecorder {
 	r.metadata = agent.metadata
 	r.logger = agent.logger
 	r.flushFrequency = agent.flushFrequency
+	r.concurrencyLevel = agent.concurrencyLevel
 	r.url = agent.getUrl("api/agent/ingest")
 	r.client = &http.Client{}
 	r.stats = &RecorderStats{}
+	r.s = semaphore.NewWeighted(int64(r.concurrencyLevel))
```

Comment:
Why not use a fixed pool of go routines that read from a channel? Like https://gobyexample.com/worker-pools

Reply:
I've found easier to implement (less code modification) with a semaphore... But yeah we can do it also with two channels... Do you prefer with a channel implementation?